src/xcode/ENA/ENA/Source/Services/Risk/KeyPackageDownload/KeyPackageDownload.swift

```diff
@@ -355,8 +356,12 @@ class KeyPackageDownload: KeyPackageDownloadProtocol {
 				case let .success(days):
 					Log.info("KeyPackageDownload: Server data is available for day packages.", log: .riskDetection)
 					completion(.success(days))
-				case .failure:
-					completion(.failure(.uncompletedPackages))
+				case .failure(let error):
+					if case .noResponse = error {
```

Comment:
Is this the correct case? There is an error .noNetworkConnection. Isn't this the correct one?

Reply:
Yes, the case .noResponse is correct. It is returned when there is no network in the failure and then we pass .noNetworkConnection which we have introduced here in KeyPackageDownload. I have tested it and it works as expected.